utils/fork-tree/src/lib.rs

```diff
@@ -319,17 +276,16 @@ where
 		F: Fn(&H, &H) -> Result<bool, E>,
 		P: Fn(&V) -> bool,
 	{
-		// search for node starting from all roots
-		for root in self.roots.iter_mut() {
-			let node = root.find_node_where_mut(hash, number, is_descendent_of, predicate)?;
-
-			// found the node, early exit
-			if let FindOutcome::Found(node) = node {
-				return Ok(Some(node))
-			}
-		}
-
-		Ok(None)
+		self.find_node_index_where(hash, number, is_descendent_of, predicate)
+			.map(|maybe_path| {
+				maybe_path.map(|path| {
+					let children = path
+						.iter()
+						.take(path.len() - 1)
+						.fold(&mut self.roots, |curr, &i| &mut curr[i].children);
+					&mut children[path[path.len() - 1]]
+				})
+			})
 	}
 
 	/// Same as [`find_node_where`](ForkTree::find_node_where), but returns indexes.
```

Comment:
```suggestion
	/// Same as [`find_node_where`](ForkTree::find_node_where), but returns indices.
	///
	/// The returned indices represent the full path to reach the matching node starting
	/// from last to first, i.e. the earliest index in the traverse path goes last, and the final 
	/// index in the traverse path goes first. If a node is found that matches the predicate
	/// the returned path should always contain at least one index, otherwise `None` is 
	/// returned.
```